src/main/java/org/fungover/mmotodo/config/SecurityConfig.java

```diff
@@ -0,0 +1,41 @@
+package org.fungover.mmotodo.config;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpMethod;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.AuthenticationSuccessHandler;
+
+@Configuration
+public class SecurityConfig {
+
+    @Value("${github.api.logout.url}")
+    private String githubSignOutUrl;
+
+    private final AuthenticationSuccessHandler authenticationSuccessHandler;
+
+    public SecurityConfig(AuthenticationSuccessHandler authenticationSuccessHandler) {
+        this.authenticationSuccessHandler = authenticationSuccessHandler;
+    }
+
+    @Bean
+    public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
+        http
+                .authorizeHttpRequests(auth -> auth
+                        .requestMatchers(HttpMethod.GET, "/auth/logout", "/api/user", githubSignOutUrl).authenticated()
+                        .requestMatchers(HttpMethod.GET, "/login").permitAll()
+                        .anyRequest().denyAll()
+                )
+                .oauth2Login(login -> login
+                        .loginPage("/login")
+                        .successHandler(authenticationSuccessHandler)
+                );
+
+        return http.build();
+    }
```

Comment:
With 'anyRequest.denyAll' you're blocking all the api endpoints. Should at the very least be changed to authenticated, probably with some public req matchers as well.